app/_src/networking/meshexternalservice.md

```diff
@@ -0,0 +1,352 @@
+This limitation will be lifted in the next release.
+{% endwarning %}
+
+## Configuration
+
+### Match
+
+This section specifies the rules for matching traffic that will be routed to external resources defined in `endpoints` section.
+The only `type` supported is [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) and it means that it will match traffic directed to a hostname created by the hostname generator.
+The `port` field when omitted means that all traffic will be matched.
+Protocols that are supported are: `tcp`, `grpc`, `http`, `http2`.
+
+```yaml
+match:
+  type: HostnameGenerator
+  port: 4244
+  protocol: tcp
+```
+
+### Endpoints
+
+This section specifies the destination of the matched traffic.
+It's possible to define IPs, DNS names and unix domain sockets.
+
+```yaml
+endpoints:
+  - address: 1.1.1.1
+    port: 12345
+  - address: example.com
+    port: 80
+  - address: unix:///tmp/example.sock
+```
+
+### TLS
+
+This section describes the TLS and verification behaviour.
+You can define TLS version requirements, option to allow renegotiation, verification of SNI, SAN, custom CA and client certificate and key for server verification.
+To disable parts of the verification you can set different `mode` - `SkipSAN`, `SkipCA`, `SkipAll`, `Secured` (default).
+
+```yaml
+tls:
+  version:
+    min: TLS12
+    max: TLS13
+  allowRenegotiation: false
+  verification:
+    mode: SkipCA
+    serverName: "example.com"
+    subjectAltNames:
+      - type: Exact
+        value: example.com
+      - type: Prefix
+        value: "spiffe://example.local/ns/local"
+    caCert:
+      inline: dGVzdA==
+    clientCert:
+      secret: "123"
+    clientKey:
+      secret: "456"
+```
+
+When TLS is enabled but `caCert` is not set, the sidecar uses the [autodetected OS-specific CA](https://github.com/kumahq/kuma/blob/aba6518fca65bc7ab52e5328eb686a51a6f98a53/app/kuma-dp/pkg/dataplane/certificate/cert.go#L12).
+The user can override the default CA by setting the path in the environment variable `KUMA_DATAPLANE_RUNTIME_DYNAMIC_SYSTEM_CA_PATH` for the sidecar.
+
+### DNS setup
+
+To be able to access `MeshExternalService` via a hostname you need to define a [HostnameGenerator](/docs/{{ page.version }}/policies/hostnamegenerator) with a `meshExternalService` selector.
+
+{% warning %}
+Do **not** hijack original addresses like httpbin.com (the way it was done with [External Service](/docs/{{ page.version }}/policies/external-services)).
+If you need to transparently pass traffic through the Mesh without modifying it use [MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough).
+{% endwarning %}
+
+For clients that the hostname is not fully known in advance please take a look at [Wildcard DNS matching in MeshPassthrough](/docs/{{ page.version }}/policies/meshpassthrough/#wildcard-dns-matching).
```

Comment:
Might be worth adding a few lines after this to add info about what this is doing internally:

- one or multiple hostname are generated by using matching hostname generators
- vip allocated
- envoy service created which will use `endpoints` as the different endpoints